src/charm.py

```diff
@@ -42,6 +45,59 @@ def __init__(self, *args):
         )
         self.framework.observe(self.on.client_relation_joined, self._client_relation_handler)
         self.framework.observe(self.on.client_relation_changed, self._client_relation_handler)
+        self.framework.observe(
+            self.on.ccr_leader_relation_joined,
+            self._on_config_changed,
+        )
+        self.framework.observe(
+            self.on.ccr_leader_relation_changed,
+            self._on_config_changed,
+        )
+        self.framework.observe(
+            self.on.ccr_follower_relation_changed,
```

Comment:
What happens on relation-broken? Do we need to make an attempt to clean up the cross-cluster replication gracefully?

Reply:
Hi Jon. Good observation. I guess that on the leader side there's not much to do if the relation is broken, but on the follower side it sounds a good idea to [stop](https://opensearch.org/docs/latest/replication-plugin/api/#stop-replication) it and than the cluster would automatically turn into standard index.
What do you think?

Reply:
I'm no expert on cross-cluster replication, I just noticed we weren't handling that case. I think in general, this charm could definitely benefit from some integration tests that deploy to microk8s, and test what happens when it is scaled up, thne down, and also when configured for CCR, then unrelated, etc. :)